plugins/analytics/public/dockerhub.py

```diff
@@ -0,0 +1,201 @@
+    def inspect_image(image, tag=None) -> dict:
+        if tag:
+            endpoint = (
+                f"https://hub.docker.com/v2/repositories/{image}/tags/{tag}/images"
+            )
+        else:
+            endpoint = f"https://hub.docker.com/v2/repositories/{image}"
+        return DockerHubApi._make_request(endpoint)
+
+    @staticmethod
+    def inspect_user(user) -> dict:
+        endpoint = f"https://hub.docker.com/v2/orgs/{user}"
+        # if orgs does not exist, redirects to
+        # https://hub.docker.com/v2/users/
+        return DockerHubApi._make_request(endpoint)
+
+    @staticmethod
+    def user_images(user, page_size=50) -> iter:
+        endpoint = f"https://hub.docker.com/v2/repositories/{user}?page_size={page_size}&ordering=last_updated"
+        yield from DockerHubApi._iter_endpoint_pages(endpoint)
+
+    # https://hub.docker.com/v2/repositories/yetiplatform/yeti/tags/2.0.1
+
+    @staticmethod
+    def image_tags(image, page_size=100) -> iter:
+        endpoint = f"https://hub.docker.com/v2/repositories/{image}/tags/?page_size={page_size}&page=1&name&ordering"
+        yield from DockerHubApi._iter_endpoint_pages(endpoint)
+
+    def image_full_details(image):
+        if image.find("/") == -1:
+            return {}
+        if image.find(":") != -1:
+            image, tag = image.split(":")
+        else:
+            tag = ""
+        image_metadata = DockerHubApi.inspect_image(image)
+        if not image_metadata:
+            return {}
+        image_metadata["user"] = DockerHubApi.inspect_user(image_metadata.get("user"))
+        image_metadata["tags"] = dict()
+        if tag:
+            image_metadata["tags"][tag] = DockerHubApi.inspect_image(image, tag)
+        else:
+            for tag in DockerHubApi.image_tags(image):
+                tag_name = tag.get("name")
+                image_metadata["tags"][tag_name] = DockerHubApi.inspect_image(
+                    image, tag_name
+                )
+        return image_metadata
+
+
+class DockerImageInspect(task.OneShotTask):
+    """DockerImageInspect analytics queries docker hub to get more
+    details related to docker_image observable.
+
+    This analytics adds several information as context to docker_image:
+    * Image stats: provide details about pulls, registered, updated
+    * Image details: provide information related to image tags including layers information and instructions
+    * User details: provides details about the user owning the image.
+
+    This analytics also creates new observables:
+    * generic observable with tag type:user_account to link to a user
+    * sha256 including image's digest, layers' digests and added files' digests
+    """
+
+    _defaults = {
+        "name": "DockerImageMetadata",
+        "description": "Fetch metadata from docker hub for a docker image",
+    }
+
+    acts_on: list[ObservableType] = [
+        ObservableType.docker_image,
+    ]
+
+    def _get_observable(self, obs_type, value):
```

Comment:
```suggestion
    def _get_or_create_observable(self, obs_type, value):
```

Reply:
Fixed in [37fa1af](https://github.com/yeti-platform/yeti/pull/904/commits/37fa1af54a10bd6f8740d966532caf5e8b050357) and [ce4996b](https://github.com/yeti-platform/yeti/pull/904/commits/ce4996bf80995751cfe9ccc88bed9e1006cfd70d)